### ISSUE TITLE
How to test a model only on images that have a certain class ? (the tools\test.py currently test the model on a fixed 5000 images regardless to the dataloader)

### ISSUE THREAD

**Issue by jawadhaidar** [open-mmlab/mmdetection#10732]
Objective:

I need to test different models per category only on the images that have this category. (coco dataset)
example: test model A on images that only have class A, test model B on images that only have class B ....

Steps:
-Adding classwise=True to the val_evaluator --> This will allow me to see APs on all categories including my sought one
-modified the config: specifically the data loaders according to the sought category --> This should filter the images and keep only the images that have the sough category.  (but it is not doing so during testing)

issue: 
tools\test.py is indeed giving the per category metrics but it is doing so on a fixed number of images (5000) that are randomly chosen rather than the images that only belong to the category.

Additional info:
I am sure that I have edited the data loaders correctly since once I use them in training, the images used to train the model for a certain category 'A' have the same number of images of category 'A' (same for other categories). The fixed 5000 images are only encountered during testing.

**Comment by hhaAndroid** [comment]
@jawadhaidar You can use a script to split the 5000 images in the COCO validation set into individual images. This way, it will be easier for you to perform your validation.
